File: Interop.Unit/UnitTest1.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace Interop.Unit
{
    [TestClass]
    public class UnitTest1
    {
        [TestMethod]
        public void InteropBuilderBuildTest()
        {
            Interop.CLI.InteropBuilder builder;
            builder = new CLI.InteropBuilder();
            Interop.CLI.InteropSettings settings;
            settings = new CLI.InteropSettings();
            settings.Name = "123123";
            StringAssert.Equals(builder.build(settings), "123123");
        }

        [TestMethod]
        public void InteropSettingsTest()
        {
            Interop.CLI.InteropSettings settings;
            settings = new CLI.InteropSettings();
            settings.Name = "test";
            StringAssert.Equals(settings.Name, "test");
        }


    }
}

File: Interop.LibUnit/unittest1.cpp
#include "stdafx.h"
#include "CppUnitTest.h"
#include "../Interop.Lib/InteropBuilder.h"
#include "../Interop.Lib/InteropSettings.h"

using namespace Microsoft::VisualStudio::CppUnitTestFramework;
using namespace Interop::System;

namespace InteropLibUnit
{		
	TEST_CLASS(InteropBuilderTest)
	{
	public:
		
		TEST_METHOD(InteropBuilderBuild)
		{
			// TODO: Your test code here
			InteropBuilder* builder;
			InteropSettings* settings;
			settings = new InteropSettings();
			settings->setName("12312");
			builder = new InteropBuilder();
			Assert::AreEqual(builder->build(settings), std::string("12312"));
		}

		TEST_METHOD(InteropSettingsTest) {
			InteropSettings* settings;
			settings = new InteropSettings();
			settings->setName("test");
			Assert::AreEqual(settings->getName(), std::string("test"));
		}

	};
}